pkg/generate/app/componentresolvers.go

```diff
@@ -197,21 +139,18 @@ type HighestUniqueScoreResolver struct {
 // Resolve resolves as the highest scored match returned by the Searcher, and
 // guarantees the match is unique (the only match with that given score)
 func (r HighestUniqueScoreResolver) Resolve(value string) (*ComponentMatch, error) {
-	matches, err := r.Searcher.Search(value)
-	if err != nil {
-		return nil, err
-	}
+	matches, err := r.Searcher.Search(true, value)
 	sort.Sort(ScoredComponentMatches(matches))
 	switch len(matches) {
 	case 0:
-		return nil, ErrNoMatch{value: value}
+		return nil, ErrNoMatch{Value: value, Errs: err}
 	case 1:
-		return matches[0], nil
+		return matches[0], errors.NewAggregate(err)
 	default:
 		if matches[0].Score == matches[1].Score {
-			return nil, ErrMultipleMatches{value, matches}
+			return nil, ErrMultipleMatches{Value: value, Matches: matches, Errs: err}
```

Comment:
should this only return(in the error object) the matches that have the same score, rather than all the matches?